lib/API/API.php

```diff
@@ -2,18 +2,18 @@
 
 namespace MailPoet\API;
 
-use MailPoetVendor\Symfony\Component\DependencyInjection\Container;
+use MailPoet\DI\ContainerWrapper;
+use MailPoetVendor\Psr\Container\ContainerInterface;
 use MailPoetVendor\Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException;
-use MailPoet\DI\ContainerFactory;
 
 if(!defined('ABSPATH')) exit;
 
 class API {
 
-  /** @var Container */
+  /** @var ContainerInterface */
```

Comment:
Free and premium containers are the instances of SymfonyContainer so I was thinking that using PSR will be better in case we want to change it. I think using this interface may also be also a restriction so that we won't add some custom methods on the DI Container.

Reply:
My suggestion doesn't make sense, I wasn't thinking about typing correctly.
What I proposed would mean a more specific type, so in order to make it work we'd have to modify the Symfony DI container itself - not what we want.

So we've agreed to retain the `MailPoetVendor\Psr\Container\ContainerInterface` interface.